src/fpm/manifest/profiles.f90

```diff
@@ -0,0 +1,879 @@
+              if (allocated(error)) return
+            end if
+          end if
+        end do
+
+        profiles_size=profiles_size+size(default_profiles)
+        allocate(profiles(profiles_size))
+        
+        do profindex=1, size(default_profiles)
+          profiles(profindex) = default_profiles(profindex)
+        end do
+
+        do iprof = 1, size(prof_list)
+          profile_name = prof_list(iprof)%key
+          call validate_compiler_name(profile_name, is_valid)
+          if (is_valid) then
+            profile_name = "all"
+            comp_list = prof_list(iprof:iprof)
+            prof_node=>table
+            call traverse_compilers(profile_name, comp_list, prof_node, error, profiles=profiles, profindex=profindex)
+            if (allocated(error)) return
+          else
+            call validate_os_name(profile_name, is_valid)
+            if (is_valid) then
+              os_list = prof_list(iprof:iprof)
+              profile_name = 'all'
+              compiler_name = DEFAULT_COMPILER
+              prof_node=>table
+              call traverse_oss(profile_name, compiler_name, os_list, prof_node, error, profiles=profiles, profindex=profindex)
+              if (allocated(error)) return
+            else
+              call get_value(table, profile_name, prof_node, stat=stat)
+              call prof_node%get_keys(comp_list)
+              call traverse_compilers(profile_name, comp_list, prof_node, error, profiles=profiles, profindex=profindex)
+              if (allocated(error)) return
+            end if
+          end if
+        end do
+
+        ! Apply profiles with profile name 'all' to matching profiles
+        do iprof = 1,size(profiles)
+          if (profiles(iprof)%profile_name.eq.'all') then
+            do profindex = 1,size(profiles)
+              if (.not.(profiles(profindex)%profile_name.eq.'all') &
+                      & .and.(profiles(profindex)%compiler.eq.profiles(iprof)%compiler) &
+                      & .and.(profiles(profindex)%os_type.eq.profiles(iprof)%os_type)) then
+                profiles(profindex)%flags=profiles(profindex)%flags// &
+                        & " "//profiles(iprof)%flags
+                profiles(profindex)%c_flags=profiles(profindex)%c_flags// &
+                        & " "//profiles(iprof)%c_flags
+                profiles(profindex)%link_time_flags=profiles(profindex)%link_time_flags// &
+                        & " "//profiles(iprof)%link_time_flags
+              end if
+            end do
+          end if
+        end do
+      end subroutine new_profiles
+
+      !> Construct an array of built-in profiles
+      function get_default_profiles(error) result(default_profiles)
+
+        !> Error handling
+        type(error_t), allocatable, intent(out) :: error
+
+        type(profile_config_t), allocatable :: default_profiles(:)
+
+        default_profiles = [ &
```

Comment:
Is `lfortran` missing?

Reply:
I think it wasn't in the `src/fpm_compiler.f90` before, but I can add it.